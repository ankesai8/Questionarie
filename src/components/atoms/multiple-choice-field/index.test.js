import React from "react";
import renderer from "react-test-renderer";

import { MultipleChoiceField } from "./index";

const getChildrenByClass = (tree, classString) => {
  const result = tree.root.find(
    (instance) => instance.props.className === classString
  ).props.value;

  return result;
};

describe("Multiple Choice Field", () => {
  it("MultipleChoiceField should render properly", () => {
    const choice = {
      label: "Yes, several months",
      value: "Yes, several months",
      selected: false,
    };
    const onChange = () => {
      return "function called";
    };


    const tree = renderer.create(
      <MultipleChoiceField
        onChange={onChange}
        choice={choice}
        checked={true}
        name="list_12110969"
      />
    );
    const value = getChildrenByClass(tree, "radio-btn");

    expect(value).toBe("Yes, several months");

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
