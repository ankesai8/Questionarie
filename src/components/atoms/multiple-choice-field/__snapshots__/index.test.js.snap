// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Choice Field MultipleChoiceField should render properly 1`] = `
<label
  className="option"
>
  <input
    checked={true}
    className="radio-btn"
    name="list_12110969"
    onChange={[Function]}
    type="radio"
    value="Yes, several months"
  />
  <span>
    Yes, several months
  </span>
</label>
`;
